wodles/aws/tests/test_aws.py

```diff
@@ -322,3 +322,27 @@ def test_config_format_created_date(date: str, expected_date: str, aws_config_bu
         Instance of the AWSConfigBucket class.
     """
     assert aws_config_bucket._format_created_date(date) == expected_date
+
+
+@pytest.mark.parametrize('log_file, expected_date', [
+    ({'Key' : 'AWSLogs/166157441623/elasticloadbalancing/us-west-1/2021/12/21/166157441623_elasticloadbalancing'}, 20211221),
+    ({'Key' : 'AWSLogs/875611522134/elasticloadbalancing/us-west-1/2020/01/03/166157441623_elasticloadbalancing'}, 20200103),
+    ({'Key' : '981837383623/iplogs/2020-09-20/2020-09-20-00-00-moyl.csv.gz'}, 20200920),
+    ({'Key' : '836629801214/iplogs/2021-01-18/2021-01-18-00-00-zxsb.csv.gz'}, 20210118),
+    ({'Key' : '2020/09/30/13/firehose_guardduty-1-2020-09-30-13-17-05-532e184c-1hfba.zip'}, 20200930),
+    ({'Key' : '2021/03/18/aws-waf-logs-delivery-stream-1-2021-03-18-10-32-48-77baca34f-efad-4f14-45bd7871'}, 20210318),
+    ({'Key' : '2021-11-12-09-11-26-B9F9F891E8D0EB13'}, 20211112)
+])
+def test_custom_get_created_date(log_file: dict, expected_date: str, aws_custom_bucket : aws_s3.AWSCustomBucket):
+    """
+    Test AWSCustomBucket's get_creation_date method.
+    Parameters
+    ----------
+    log_file : dict
+        The log file introduced
+    expected_date : str
+        The date that the method should return.
+    aws_custom_bucket : aws_s3.AWSCustomBucket
+        Instance of the AWSCustomBucket class.  
+    """
+    assert aws_custom_bucket.get_creation_date(log_file) == expected_date
```

Comment:
Even though the different classes you are testing inherit from `AWSCustomBucket` you shouldn't use a `AWSCustomBucket` instance to test them all.
The point of the tests is to be in an environment as close to reality as possible to make sure that whatever you are testing works in a real situation. If any of those classes modified the regex the test would pass, but the functionality would be wrong. Please, change this accordingly and make a **thorough** unit test.
Also, this test doesn't consider the case in which there are no matches in the string.


Finally, I think that it might be interesting to add more test cases.